WordPress/wordpress-playground
# Use Requests 2.0+ namespaced interfaces when available.

## What is this PR doing?

Uses the WP 6.2+ / Requests 2.0+ namespaced classes, to avoid deprecation warnings, while supporting older clients.

## What problem is it solving?

Fixes #922 PHP deprecation notices, which can be seen in [Theme Directory live previews](https://themes.trac.wordpress.org/ticket/159444#:~:text=zip%3Fnostats%3D1-,Live%20preview,-%E2%80%93%20https%3A//playground)

## How is the problem addressed?

Conditionally selecting the class to use

## Testing Instructions

1. Go to http://localhost:5400/website-server/#%7B%22preferredVersions%22:{%22wp%22:%22latest%22,%22php%22:%227.4%22},%22steps%22:%5B%7B%22step%22:%22defineWpConfigConsts%22,%22consts%22:%7B%22WP_DEBUG%22:true%7D%7D%5D%7D 
2. Confirm the following deprecation notices are not shown:

```
Deprecated: The PSR-0 'Requests_... class names in the Requests library are
deprecated. Switch to the PSR-4 'WpOrg|Requests|.. class names at your
```

(Other deprecation notices are expected as they're present in WordPress core)
